main.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/ONSdigital/dp-api-clients-go/renderer"
+	health "github.com/ONSdigital/dp-healthcheck/healthcheck"
+
+	"github.com/ONSdigital/dp-frontend-homepage-controller/config"
+	"github.com/ONSdigital/dp-frontend-homepage-controller/routes"
+	"github.com/ONSdigital/go-ns/server"
+	"github.com/ONSdigital/log.go/log"
+	"github.com/gorilla/mux"
+)
+
+var (
+	// BuildTime represents the time in which the service was built
+	BuildTime string
+	// GitCommit represents the commit (SHA-1) hash of the service that is running
+	GitCommit string
+	// Version represents the version of the service that is running
+	Version string
+)
+
+func main() {
+	ctx := context.Background()
+	if err := run(ctx); err != nil {
+		log.Event(ctx, "unable to run application", log.Error(err))
+		os.Exit(1)
+	}
+}
+
+func run(ctx context.Context) error {
+	log.Namespace = "dp-frontend-homepage-controller"
+
+	signals := make(chan os.Signal, 1)
+	signal.Notify(signals, syscall.SIGINT, syscall.SIGTERM)
+
+	cfg, err := config.Get()
+	if err != nil {
+		log.Event(ctx, "unable to retrieve service configuration", log.Error(err))
+		return err
+	}
+
+	log.Event(ctx, "got service configuration", log.Data{"config": cfg})
+
+	versionInfo, err := health.NewVersionInfo(
+		BuildTime,
+		GitCommit,
+		Version,
+	)
+
+	r := mux.NewRouter()
+
+	rend := renderer.New(cfg.RendererURL)
+
+	healthcheck := health.New(versionInfo, cfg.HealthCheckCriticalTimeout, cfg.HealthCheckInterval)
+	if err = registerCheckers(ctx, &healthcheck, rend); err != nil {
+		return err
+	}
+	routes.Init(ctx, r, healthcheck, rend)
+
+	healthcheck.Start(ctx)
+
+	s := server.New(cfg.BindAddr, r)
+	s.HandleOSSignals = false
+
+	log.Event(ctx, "Starting server", log.Data{"config": cfg})
+
+	go func() {
+		if err := s.ListenAndServe(); err != nil {
+			log.Event(ctx, "failed to start http listen and serve", log.Error(err))
+			return
+		}
+	}()
+
+	for {
+		select {
+		case <-signals:
+			log.Event(nil, "os signal received")
+			return gracefulShutdown(cfg, s, healthcheck)
+		}
+	}
```

Comment:
For loop isn't needed here. The code will block until one signal is returned then it will return for the func so there will never be another iteration of the loop. The following return will therefore never be reached. In fact the case isn't needed as we're only ever waiting for one channel.

Reply:
As @jessjenkins has pointed out this loop is definitely not needed.  With regards to the `select` it is also not needed in this case, as Jess has said it is not needed unless there is more than one channel (as is the case in some of our other apps).

The above can be replaced with the simpler:
```
<-signals:
log.Event(nil, "os signal received")
return gracefulShutdown(cfg, s, healthcheck)
```